### DISCUSSION TITLE
RPC function not returning all data in Javascript app?

### DISCUSSION THREAD

**Discussion by old-rob** [supabase/supabase#26686] [Questions] [answered]
For context, my database has three important things: items, categories, and clicks. My end goal is for my javascript to be able to call my database with a parameter for categories to include, and a parameter for start and end dates to consider clicks from. This would return a list of any item with a category that matches the first parameter, and the clicks for that item would be a count of every click that was created between the two provided dates.

The problem is that my RPC function refuses to send the categories along to my javascript. Here are some screenshots to show the issue. I've tried so many things to get it to work, but I need some advice. I'm very new to supabase and database design.

<img width="833" alt="Screenshot 2024-05-22 at 3 28 01 PM" src="https://github.com/supabase/supabase/assets/42311776/865c083c-768a-4ee0-8c95-116dcf41cf4d">
<img width="1117" alt="Screenshot 2024-05-22 at 3 28 55 PM" src="https://github.com/supabase/supabase/assets/42311776/840bf930-eb37-4439-878c-9d7c404ce039">
<img width="711" alt="Screenshot 2024-05-22 at 4 18 09 PM" src="https://github.com/supabase/supabase/assets/42311776/2ad94b31-98ef-4fa4-8591-a5e175d31886">

This is how I'm calling the function: 
`const { data, error } = await supabase.rpc('rpc_test', { "json_params": {
      "categories": '{"colors", "test"}',
      "start_date": "2024-04-09 00:22:35.837547+00"
    } })
    `

I've seen various levels of success, but I've never been able to get it all to work together.

**Comment by GaryAustin1** [comment]
Do you have RLS enabled for all the tables for the user to access?
Does it work if you run it in the SQL editor (there is also impersonation option there to test different user access).

**Comment by silentworks** [comment, marked as answer]
I can't see some of your SQL here but the table you are getting the categories from is likely to have row level security (RLS) turned on and you are trying to make a request to that table without sufficient permission. A easy way to test this is inside of your SQL editor click on `role postgres` and change to `authenticated` and then select a user you want to run this query. Also try anon role too and see if the results match your `.rpc` function call.
![image](https://github.com/supabase/supabase/assets/79497/ec485b1b-bb82-4006-8a7b-44227c4c32f1)
